src/utils/search.js

```diff
@@ -18,13 +18,15 @@ export let newNodeFromType = _.curry((type, fields, node) => ({
   ...defaultNodeProps(node.field, fields, type),
 }))
 
-// In the future, we will have logic here to handle intelligently updating
-// a node's types when its field is mutated. When that happens, we'll also
-// need to pass the `fields` schema to look up typeOptions for the current
-// and future field. Could also be a node transformer like newNodeFromType
-// if we decide to use Tree.replace() for fields instead of Tree.mutate().
-export let changeNodeField = (Tree, node, field) => {
-  Tree.mutate(node.path, { field })
+export let newNodeFromField = ({ key, field, fields, ...rest }) => {
```

Comment:
personally, I don't prefer this style of argument passing unless it's really necessary - I'd probably just have done `(field, fields, rest = {})`, with `key` in the rest object (or not at all, see the next comment)

it's really just a matter of taste, though, so it's up to you 🙂